src/OrchardCore/OrchardCore.OpenId.Core/YesSql/Migrations/OpenIdMigrations.cs

```diff
@@ -80,5 +80,67 @@ public int UpdateFrom2()
 
             return 3;
         }
+
+        public int UpdateFrom3()
+        {
+            SchemaBuilder.AlterTable(nameof(OpenIdApplicationIndex), table =>
+            {
+                table.CreateIndex($"IX_{nameof(OpenIdApplicationIndex)}_{nameof(OpenIdApplicationIndex.ApplicationId)}", nameof(OpenIdApplicationIndex.ApplicationId));
+                    table.CreateIndex($"IX_{nameof(OpenIdApplicationIndex)}_{nameof(OpenIdApplicationIndex.ClientId)}", nameof(OpenIdApplicationIndex.ClientId));
+            });
+            SchemaBuilder.AlterTable(nameof(OpenIdAppByLogoutUriIndex), table =>
+            {
+                table.CreateIndex($"IX_{nameof(OpenIdAppByLogoutUriIndex)}_{nameof(OpenIdAppByLogoutUriIndex.LogoutRedirectUri)}", nameof(OpenIdAppByLogoutUriIndex.LogoutRedirectUri));
+            });
+            SchemaBuilder.AlterTable(nameof(OpenIdAppByRedirectUriIndex), table =>
+            {
+                table.CreateIndex($"IX_{nameof(OpenIdAppByRedirectUriIndex)}_{nameof(OpenIdAppByRedirectUriIndex.RedirectUri)}", nameof(OpenIdAppByRedirectUriIndex.RedirectUri));
+            });
+            SchemaBuilder.AlterTable(nameof(OpenIdAppByRoleNameIndex), table =>
+            {
+                table.CreateIndex($"IX_{nameof(OpenIdAppByRoleNameIndex)}_{nameof(OpenIdAppByRoleNameIndex.RoleName)}", nameof(OpenIdAppByRoleNameIndex.RoleName));
+            });
+            SchemaBuilder.AlterTable(nameof(OpenIdAuthorizationIndex), table =>
+            {
+                table.CreateIndex($"IX_{nameof(OpenIdAuthorizationIndex)}_{nameof(OpenIdAuthorizationIndex.Subject)}", nameof(OpenIdAuthorizationIndex.Subject));
```

Comment:
Someone needs to check the lengths of these names against postgresql, as we have had issues there before.

@jptissot can you comment?

Reply:
I've checked migrations runs properly at sqlite, sql server and postgresql.
Also I've created openid applications and reguested access and refresh token without problem with postgresql and sqlite.
It is enough, or need any extra check?

Reply:
PostgreSQL 12

Reply:
MySql is the one that is most likely to fail, so we do need to check that. (sorry)

It's generally an exception on migration that occurs with mysql.

Great to have checked postgresql thanks.

Reply:
ok, I will install it

Reply:
You were right 🥇 
There was two indexes that where too much long for MySQL.
I've commited a change that doesn't apply those indexes when dialect is MySql

Great job

Reply:
I only know because I broke MySql for 3 months ;)

So chances are if the indexes are too long for MySql they are also too long for MS SQL.
The only difference is MS SQL gives you a warning, instead of an exception, and then throws an exception when you try to update the field with a value that is too long.

(See here for how we had to manage it for the sql fields index https://github.com/OrchardCMS/OrchardCore/pull/5357/files)

The way to know if MS SQL is going to have the same potential issue, is to use SQL Management Studio to drop and create the index, you might get a warning, like this 

`Warning! The maximum key length for a nonclustered index is 1700 bytes. The index 'IDX_AliasPartIndex_Alias' has maximum length of 3001 bytes. For some combination of large values, the insert/update operation will fail.`